core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/CompactEntityComponent.scala

```diff
@@ -160,25 +215,41 @@ class CompactEntityQuery(driverComponent: DriverComponent) extends RawSqlQuery w
     *
     * `execution plan: batched insert (one statement, multiple rows)`
     */
-  def upsertReferences(fromId: Long, toIds: Set[Long]): ReadWriteAction[Int] = {
-    val insertValues: Iterable[SQLActionBuilder] = toIds.map { toId =>
-      sql"($fromId,$toId)"
-    }
-    val allInsertValues = reduceSqlActionsWithDelim(insertValues.toSeq, sql",")
+  def upsertReferences(fromId: Long, toIds: Set[Long]): ReadWriteAction[Int] =
+    upsertReferences(Set((fromId, toIds)))
 
-    val query = concatSqlActions(
-      sql"""insert into ENTITY_REFS(from_id, to_id)
```

Comment:
I added an upsertReferences method which accepts multiple "from" ids, and rewrote the existing `upsertReferences(Long, Set[Long)` to delegate to it.